### PR TITLE
BEI-6: Setup linting with Eslint

### PR DESCRIPTION
**Pull request by hizikex**
#### WHY THIS PR?
- There is a need identify and report common code issues, enforcing coding styles and improve code quality.

#### WHAT DOES THIS PR DO?
-  It helps to catch common code issues.
-  It helps to maintain consistent coding style.
- It improves cod readability.

HOW CAN THIS PR BE TESTED?
- Create a .eslintrc.js file in the root directory, and map {"extends": "airbnb-typescript"}

THINGS YOU MAY WANT THE REVIEWER TO NOTE.
N/A

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+          // warns when it encounters a reference to an identifier that has not yet been declared.
+          'error',
+          {
+            functions: true,
+            classes: true,
+            variables: false
+          }
+        ],
+    }
+  };
```
[the remark this answers is not in the document]
Comment (hizikex) [reply .eslintrc.js:37]: Thanks.

**Review by LucianaDera — APPROVED** [review]

**Review by RICHYDOS — APPROVED** [review]

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
```
Comment (LordUche) [inline .eslintrc.js:38]:
```suggestion
  };

```
There seems to be some extra space. Kindly take them out.

**Review by LordUche — CHANGES_REQUESTED** [review]

**Comment by LordUche** [comment]
@hizikex Hound CI is complaining about the eslintrc file format. Apparently, it only supports JSON and Yaml. You can change it to a JSON file by renaming it to `.eslintrc.json` or `.eslintrc` and then changing the contents of the file to JSON instead of JavaScript.

Also, remember to update the `.hound.yml` file to point to the correct eslintrc file.

File: .eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
+{
+    "env": {
+        "commonjs": true,
+        "es2021": true,
+        "node": true
+    },
+    "extends": "airbnb-typescript",
+    "parserOptions": {
+        "ecmaVersion": "latest"
```
Comment (LordUche) [inline .eslintrc.json:9]:
```suggestion
        "ecmaVersion": "latest",
        "sourceType": "module"
```

**Review by LordUche — CHANGES_REQUESTED** [review]

File: .eslintrc.json (hunk 3 above)
Comment (hizikex) [reply .eslintrc.json]: This has been done.

**Review by LordUche — APPROVED** [review]

**Pull request merged**